pkg/controller/namespace/namespace_controller.go

```diff
@@ -157,20 +157,21 @@ func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Res
 			}
 		} else {
 			// Don't need injection, may be need to remove the sidecar?
-			// If deployment don't have the annotation and has an agent, this may be injected by the namespace
+			// If deployment don't have the annotation and has an hasAgent, this may be injected by the namespace
 			// we need to clean it.
-			agent, _ := inject.HasJaegerAgent(dep)
-			_, depAnnotation := dep.Annotations[inject.Annotation]
-			if agent && !depAnnotation{
-				jaegerInstance, label := dep.Labels[inject.Label]
-				if label {
+			hasAgent, _ := inject.HasJaegerAgent(dep)
+			_, hasDepAnnotation := dep.Annotations[inject.Annotation]
+			if hasAgent && !hasDepAnnotation {
+				jaegerInstance, hasLabel := dep.Labels[inject.Label]
+				if hasLabel {
 					log.WithFields(log.Fields{
 						"deployment":       dep.Name,
 						"namespace":        dep.Namespace,
 						"jaeger":           jaegerInstance,
 					}).Info("Removing Jaeger Agent sidecar")
+					patch := client.MergeFrom(dep.DeepCopy())
```

Comment:
If `dep` is a pointer, you might not need to do a DeepCopy, you can just dereference it.